pkg/skaffold/runner/deploy_test.go

```diff
@@ -42,34 +42,54 @@ import (
 func TestDeploy(t *testing.T) {
 	expectedOutput := "Waiting for deployments to stabilize..."
 	tests := []struct {
-		description string
-		testBench   *TestBench
-		statusCheck config.BoolOrUndefined
-		shouldErr   bool
-		shouldWait  bool
+		description    string
+		testBench      *TestBench
+		statusCheckCLI config.BoolOrUndefined // --status-check CLI flag
+		statusCheck    bool                   // skaffold.yaml Deploy.StatusCheck field
+		shouldErr      bool
+		shouldWait     bool
 	}{
 		{
-			description: "deploy shd perform status check",
-			testBench:   &TestBench{},
-			statusCheck: config.NewBoolOrUndefined(nil),
-			shouldWait:  true,
+			description:    "deploy shd perform status check when statusCheckCLI true, statusCheck true",
+			testBench:      &TestBench{},
+			statusCheckCLI: config.NewBoolOrUndefined(util.BoolPtr(true)),
+			statusCheck:    true,
+			shouldWait:     true,
 		},
 		{
-			description: "deploy shd perform status check",
-			testBench:   &TestBench{},
-			statusCheck: config.NewBoolOrUndefined(util.BoolPtr(true)),
-			shouldWait:  true,
+			description:    "deploy shd perform status check when statusCheckCLI true, statusCheck false",
+			testBench:      &TestBench{},
+			statusCheckCLI: config.NewBoolOrUndefined(util.BoolPtr(true)),
+			shouldWait:     true,
 		},
 		{
-			description: "deploy shd not perform status check",
-			testBench:   &TestBench{},
-			statusCheck: config.NewBoolOrUndefined(util.BoolPtr(false)),
+			description:    "deploy shd not perform status check when statusCheckCLI false, statusCheck true",
```

Comment:
how about: `when statusCheck flag is false, in-config value is true`

Reply:
Much cleaner, thanks!